org.compiere.db.oracle.provider/src/org/adempiere/db/oracle/partition/TablePartitionService.java

```diff
@@ -949,4 +952,51 @@ public void reattachPartition(MTable table, X_AD_TablePartition partition, Strin
 		}
 	}
 
+	/**
+	 * Get default partition name for table
+	 * @param table
+	 * @return String default partition name for table
+	 */
+	public String getDefaultPartitionName(MTable table)
+	{
+		return getDefaultPartitionName(table.getTableName());
+	}
+	
+	@Override
+	public String getDefaultPartitionName(String tableName) {
+		return tableName + "_default_partition";
+	}
+	
+	@Override
+	public String parsePartitionName(String tableName, Object[] partitionCols) {
+		StringBuilder partitionName = new StringBuilder();
+		if(Util.isEmpty(tableName))
+			return "";
+		
+		partitionName.append(tableName);
+		for(Object partitionCol : partitionCols) {
+			String sPartitionCol = Objects.toString(partitionCol);
+			if(!Util.isEmpty(sPartitionCol)) {
+				partitionName.append("_").append(sPartitionCol);
+			}
+		}
+		
+		return partitionName.toString();
+	}
```

Comment:
If we were to add this API, it should be extracted from the refactoring of existing code. This duplication of code is unsafe and add unnecessary maintenance burden.

Reply:
Extracted the duplicated methods into TablePartitionUtils.java